lib/mongo_trails/model_config.rb

```diff
@@ -14,5 +14,40 @@ def #{@model_class.versions_association_name}
         end
       RUBY
     end
+
+    def on_create
+      @model_class.after_create_commit do |r|
+        r.paper_trail.record_create if r.paper_trail.save_version?
+      end
+      append_option_uniquely(:on, :create)
+    end
+
+    def on_destroy(recording_order = 'before')
+      assert_valid_recording_order_for_on_destroy(recording_order)
+      @model_class.after_destroy_commit do |r|
+        break unless r.paper_trail.save_version?
+        r.paper_trail.record_destroy(recording_order)
+      end
+      append_option_uniquely(:on, :destroy)
+    end
+
+    def on_update
+      @model_class.before_save do |r|
+        r.paper_trail.reset_timestamp_attrs_for_update_if_needed
+      end
+      @model_class.after_update_commit do |r|
+        if r.paper_trail.save_version?
+          r.paper_trail.record_update(
```

Comment:
will this track all the creates and updates inside a single transaction?
maybe use AfterCommitEverywhere gem instead?